madmom/io/__init__.py

```diff
@@ -333,62 +345,67 @@ def write_notes_mirex(notes, filename, duration=0.6):
     return notes
 
 
-def load_chords(filename):
+def load_segments(filename):
     """
-    Load labelled chord segments from a file. Chord segments must follow
-    the following format, one chord label per line:
-
-    <start_time> <end_time> <chord_label>
-
-    All times should be given in seconds.
+    Load labelled segments from a text file. Each line in the file represents
+    a segment and is of the form <start> <end> <label>, where <start> and
+    <end> are floating point numbers, and <label> is a string.
 
     Parameters
     ----------
     filename : str or file handle
-        File containing the segments
+        File containing labelled segments.
 
     Returns
     -------
-    numpy structured array
-        Structured array with columns 'start', 'end', and 'label', containing
-        the start time, end time, and segment label respectively
+    segments : numpy structured array
+        Structured array with columns "start", "end", and "label",
+        containing the beginning, end, and label of segments.
+    """
+    start, end, label = [], [], []
 
-    Notes
-    -----
-    Segment files cannot contain comments, because e.g. chord annotations
-    can contain the '#' character! The maximum label length is 32 characters.
+    def open_file(fn):
+        return fn if isinstance(fn, _file_handle) else open(fn, 'r')
 
-    """
-    from ..features.chords import CHORD_DTYPE
-    return np.loadtxt(filename, comments=None, ndmin=1, dtype=CHORD_DTYPE,
-                      converters={2: lambda x: x.decode()})
+    with open_file(filename) as f:
+        for line in f:
+            s, e, l = line.split()
+            start.append(float(s))
+            end.append(float(e))
+            label.append(l)
+
+    segments = np.zeros(len(start), dtype=SEGMENT_DTYPE)
+    segments['start'] = start
+    segments['end'] = end
+    segments['label'] = label
+    return segments
```

Comment:
Ok, just asking.

I once also did the loading of events manually, but decided to trade speed for flexibility. E.g. `np.loadtxt` ignores empty lines, can handle basically every possible separator, has nice handling for comments (in case of chords it is perfectly working with `comments='# '`), etc...

Your decision, though.